### ISSUE TITLE
400 (Bad Request)

### ISSUE THREAD

**Issue by GhermanTeodora** [ed-roh/mern-social-media#7]
Hello! When I try to log in I get this error: "POST http://localhost:3001/auth/login 400 (Bad Request) Form.jsx:82". I did not change any of the backend code and I connected to mongodb, I even put your predefined users and posts in collections and tried one of them to log in and it is not working. What should I do?

**Comment by smallbitmatters** [comment]
Hi, Can you try to your mongodb Atlas>database tab>overview
Click for the connect button and this image will pops up. after that click  the "Connect your application"
Then when you copy/paste the link into your .env file, make sure to remove " password including the greater/lesser than " and place your password there.
mongodb+srv://YOURMONGODBATLASUSERNAMEFROMYOURCLUSTER:<password>@CLUSTERNAME.RANDOMCHAR.mongodb.net/?retryWrites=true&w=majority
![image](https://user-images.githubusercontent.com/51928225/207472418-909b74fb-f5c0-4bb9-a4c7-6d0f4dcb1b90.png)

**Comment by GhermanTeodora** [comment]
Yes, I have already done this.

**Comment by pikaden** [comment]
> Yes, I have already done this.

![image](https://user-images.githubusercontent.com/87664605/209825706-2e17dbd2-c84b-4e4c-a708-f9e014077216.png)
try copy .env to server folder

**Comment by Kamilm55** [comment]
Can you run it?

**Comment by Pukhraj85** [comment]
> Hello! When I try to log in I get this error: "POST http://localhost:3001/auth/login 400 (Bad Request) Form.jsx:82". I did not change any of the backend code and I connected to mongodb, I even put your predefined users and posts in collections and tried one of them to log in and it is not working. What should I do?

Maybe you are not running the backend server. See if your backend server is also running

**Comment by KjOHnBaggins** [comment]
I'm also getting the same **400** error with **login**. But I noticed something, that when I register, the img I uploaded gets into the server/public folder and the commad line of the server is responding to every single submit I make. It's just that I'm getting the **404** for my **register**(the img still goes thru tho).

**Comment by KjOHnBaggins** [comment]
I've solved the problem  somehow :) . This is a routing problem.  In the routes/auth.js file, I added this. `router.post("/register", register);`  Now, when I register, it takes me to the login, I fill the login and got to the home page. I don't know it's right or not, I'm still a beginner at express, but it sort of solve the problem right now. Hope this help!

**Comment by Kamilm55** [comment]
> I've solved the problem somehow :) . This is a routing problem. In the routes/auth.js file, I added this. `router.post("/register", register);` Now, when I register, it takes me to the login, I fill the login and got to the home page. I don't know it's right or not, I'm still a beginner at express, but it sort of solve the problem right now. Hope this help!

do you add your own database with mongo?

**Comment by KjOHnBaggins** [comment]
> do you add your own database with mongo?

Yes, I did. And the data I filled in the register page goes into the collection, inside the test/user. So I guess it's kind of working? 
  ¯\ _ (ツ)_/¯

**Comment by Kamilm55** [comment]
> > Hello! When I try to log in I get this error: "POST http://localhost:3001/auth/login 400 (Bad Request) Form.jsx:82". I did not change any of the backend code and I connected to mongodb, I even put your predefined users and posts in collections and tried one of them to log in and it is not working. What should I do?
> 
> Maybe you are not running the backend server. See if your backend server is also running

I don't know more about backend i write localhost:3001 and i saw my backend not working how can i run backend

**Comment by KjOHnBaggins** [comment]
> I don't know more about backend i write localhost:3001 and i saw my backend not working how can i run backend

It is very simple. I assume you have node installed.

1. Open terminal in your VS code.
2. Navigate to the location of your server folder.
3. run `node index.js`

This will run your server. You don't need to see or open the port of your backend in browser, you just run the command. Then you register in your frontend, and when you click the register button, your server terminal should respond. Hope this helps!

**Comment by Kamilm55** [comment]
> > I don't know more about backend i write localhost:3001 and i saw my backend not working how can i run backend
> 
> It is very simple. I assume you have node installed.
> 
> 1. Open terminal in your VS code.
> 2. Navigate to the location of your server folder.
> 3. run `node index.js`
> 
> This will run your server. You don't need to see or open the port of your backend in browser, you just run the command. Then you register in your frontend, and when you click the register button, your server terminal should respond. Hope this helps!

I registered it worked and when i login it says Status Code: 204 No Content and the following it says Status Code: 500 Internal Server Error

**Comment by KjOHnBaggins** [comment]
Hold on, have you created your own database on mongodb cluster?

**Comment by KjOHnBaggins** [comment]
> I registered it worked and when i login it says Status Code: 204 No Content and the following it says Status Code: 500 Internal Server Error

Hold on, have you created your own mongodb database on mongodb clusters?

**Comment by Kamilm55** [comment]
Yes

**Comment by Kamilm55** [comment]
> > I registered it worked and when i login it says Status Code: 204 No Content and the following it says Status Code: 500 Internal Server Error
> 
> Hold on, have you created your own mongodb database on mongodb clusters?

Yes

**Comment by KjOHnBaggins** [comment]
> > Hold on, have you created your own mongodb database on mongodb clusters?
> 
> Yes

I guess this would be MongoDB cluster malfunction. Here's a couple methods I suggest.

- Try reloading your cluster.
- Adding your IP address to the cluster.
- Register again and check in the browse collections tab.
- Recreating another cluster or project.
